### PR TITLE
scroll to the error message when it's not visible

### PR DESCRIPTION
**Pull request by achour94**

### CODE REVIEW

**Review by souissimai — CHANGES_REQUESTED** [review]
please fix prettiers

File: src/components/utils/rhf-inputs/error-inputs/error-input.js [hunk 1]
```diff
@@ -16,6 +16,10 @@ const ErrorInput = ({ name, InputField }) => {
         name,
     });
 
+    const { isSubmitted } = useFormState();
```
Comment (flomillot) [inline src/components/utils/rhf-inputs/error-inputs/error-input.js:19]: No need this, add 
`formState: { isSubmitted },`
inside the `useController`

File: src/components/utils/rhf-inputs/error-inputs/error-input.js [hunk 2]
```diff
@@ -32,11 +36,19 @@ const ErrorInput = ({ name, InputField }) => {
         return {};
     };
 
+    useEffect(() => {
+        if (isSubmitted && error && errorRef?.current) {
```
Comment (flomillot) [inline src/components/utils/rhf-inputs/error-inputs/error-input.js:40]: Why `?` on `errorRef`, how can it be null ?

File: src/components/utils/rhf-inputs/error-inputs/error-input.js (hunk 1 above)
Comment (flomillot) [reply src/components/utils/rhf-inputs/error-inputs/error-input.js:19]: And why using `isSubmitted` instad of `isSubmitting` ?
Then it focus everytime you submit

Comment (achour94) [reply src/components/utils/rhf-inputs/error-inputs/error-input.js:19]: if you use isSubmitting you risk to trigger the effect before the validation is done, i added error to the useEffect dependencies so it scroll each time you submit

File: src/components/utils/rhf-inputs/error-inputs/error-input.js (hunk 2 above)
Comment (achour94) [reply src/components/utils/rhf-inputs/error-inputs/error-input.js:40]: because in the first render the errorRef will be null

Comment (flomillot) [reply src/components/utils/rhf-inputs/error-inputs/error-input.js:39]: Current will be null, not errorRef

**Review by flomillot — APPROVED** [review]

**Review by souissimai — APPROVED** [review]

**Pull request closed without merging**